src/snowflake/snowpark/session.py

```diff
@@ -491,6 +510,19 @@ def sql_simplifier_enabled(self) -> bool:
         """
         return self._sql_simplifier_enabled
 
+    @sql_simplifier_enabled.setter
+    def sql_simplifier_enabled(self, value: bool) -> None:
+        self._conn._telemetry_client.send_sql_simplifier_telemetry(
+            str(self._session_id), value
+        )
+        try:
+            self._conn._cursor.execute(
+                f"alter session set {_PYTHON_SNOWPARK_USE_SQL_SIMPLIFIER_STRING} = {value}"
+            )
+        except Exception:
```

Comment:
Could use [contextlib.suppress](https://docs.python.org/3/library/contextlib.html#contextlib.suppress) for this